lib/ansible/module_utils/basic.py

```diff
@@ -2498,6 +2501,48 @@ def preserved_copy(self, src, dest):
         current_attribs = current_attribs.get('attr_flags', '')
         self.set_attributes_if_different(dest, current_attribs, True)
 
+    def lock_file(self, filename, lock_timeout=0):
+        if self.check_mode:
+            return True
+
+        self.lockfd = None
+        tmpdir = None
+        if not tmpdir:
+            tmpdir = tempfile.gettempdir()
+
+        lock_file = os.path.join(tmpdir, 'ansible-{0}.lock'.format(os.path.basename(filename)))
+        l_wait = 0.2
+        if lock_timeout > 0:
+            e_secs = 0
+            while e_secs < lock_timeout:
+                try:
+                    self.lockfd = open(lock_file, 'w')
```

Comment:
This seems like it will make for a hard API to use because it will fail when the lock_file is owned by another user (so playbooks run by different users or async with tasks that become different users will raise Permission denied errors).  It seems like problems opening the lock_file should be part of the timeout.